tests/integration/large_deployments/test_orchestration.py

```diff
@@ -0,0 +1,353 @@
+from ..tls.test_tls import TLS_CERTIFICATES_APP_NAME, TLS_STABLE_CHANNEL
+
+logger = logging.getLogger(__name__)
+MAIN_APP = "opensearch-main"
+FAILOVER_APP = "opensearch-failover"
+DATA_APP = "opensearch-data"
+_2nd_DATA_APP = "opensearch-data-two"
+
+CLUSTER_NAME = "app"
+
+APP_UNITS = {MAIN_APP: 1, FAILOVER_APP: 1, DATA_APP: 1, _2nd_DATA_APP: 1}
+
+CMS_NOT_ENOUGH_STATUS = (
+    "Less than 3 cluster-manager-eligible units in this cluster. Add more units."
+)
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.skip_if_deployed
+async def test_build_and_deploy(ops_test: OpsTest, charm, series) -> None:
+    """Build and deploy one unit of OpenSearch."""
+    assert ops_test.model is not None
+    await ops_test.model.set_config(MODEL_CONFIG)
+
+    await ops_test.model.create_storage_pool("local", "lxd", "volume-type=standard")
+
+    # Deploy TLS Certificates operator.
+    config = {"ca-common-name": "CN_CA"}
+    await asyncio.gather(
+        ops_test.model.deploy(
+            TLS_CERTIFICATES_APP_NAME, channel=TLS_STABLE_CHANNEL, config=config
+        ),
+        ops_test.model.deploy(
+            charm,
+            application_name=MAIN_APP,
+            num_units=APP_UNITS[MAIN_APP],
+            series=series,
+            config={"cluster_name": CLUSTER_NAME, "roles": "cluster_manager,data"} | CONFIG_OPTS,
+            storage={"opensearch-data": "local,128G,1"},
+        ),
+        ops_test.model.deploy(
+            charm,
+            application_name=FAILOVER_APP,
+            num_units=APP_UNITS[FAILOVER_APP],
+            series=series,
+            config={"cluster_name": CLUSTER_NAME, "roles": "cluster_manager", "init_hold": True}
+            | CONFIG_OPTS,
+            storage={"opensearch-data": "local,128G,1"},
+        ),
+        ops_test.model.deploy(
+            charm,
+            application_name=DATA_APP,
+            num_units=APP_UNITS[DATA_APP],
+            series=series,
+            config={"cluster_name": CLUSTER_NAME, "roles": "data", "init_hold": True}
+            | CONFIG_OPTS,
+            storage={"opensearch-data": "local,128G,1"},
+        ),
+        ops_test.model.deploy(
+            charm,
+            application_name=_2nd_DATA_APP,
+            num_units=APP_UNITS[_2nd_DATA_APP],
+            series=series,
+            config={"cluster_name": CLUSTER_NAME, "roles": "data", "init_hold": True}
+            | CONFIG_OPTS,
+            storage={"opensearch-data": "local,128G,1"},
+        ),
+    )
+    for app in APP_UNITS:
+        await ops_test.model.integrate(app, TLS_CERTIFICATES_APP_NAME)
+
+    for app in [FAILOVER_APP, DATA_APP, _2nd_DATA_APP]:
+        await ops_test.model.integrate(
+            f"{MAIN_APP}:peer-cluster-orchestrator", f"{app}:peer-cluster"
+        )
+
+    for app in [DATA_APP, _2nd_DATA_APP]:
+        await ops_test.model.integrate(
+            f"{FAILOVER_APP}:peer-cluster-orchestrator", f"{app}:peer-cluster"
+        )
+
+    await wait_until(
```

Comment:
I think we should increase timeout here since I got a timeout when trying on my setup